lib/search/search.ts

```diff
@@ -298,56 +303,56 @@ export abstract class AbstractSearch {
    * @returns An array of key names matching the query.
    */
   async allKeys(options = { pageSize: 10 }): Promise<string[]> {
-    return this.allThings(this.pageOfKeys, options) as Promise<string[]>
+    return await this.allThings(this.pageOfKeys, options) as string[]
```

Comment:
I think I was having typescript battles. Looking at it, if `allThings` is a promise, you cant cast it to a string[] unless you wrap the await and the function call with parentheses. 

Why I didn't await it in this case may be an issue with my IDE recognizing it as a promise, and I missed the await.

Reply:
Ya. Should work regardless. I was just scanning for consistency and wondered if there was a reason.

Reply:
Ah yes, I remember now, I wasn't sure about the internal `allThings` function since it could return either `T[] | string[]`, so I updated the typing for the function so it can correctly infer the return type of `allThings` depending on which `pageFn` is being used.